cmd/gpu_plugin/README.md

```diff
@@ -205,6 +218,31 @@ GPU plugin does __not__ setup SR-IOV. It has to be configured by the cluster adm
 
 GPU plugin does however support provisioning Virtual Functions (VFs) to containers for a SR-IOV enabled GPU. When the plugin detects a GPU with SR-IOV VFs configured, it will only provision the VFs and leaves the PF device on the host.
 
+### KMD and UMD
+
+There are 3 different Kernel Mode Drivers (KMD) available: `i915 upstream`, `i915 backport` and `xe`:
+* `i915 upstream` is a vanilla driver that comes from the upstream kernel and is included in the common Linux distributions, like Ubuntu.
+* `i915 backport` is an [out-of-tree driver](https://github.com/intel-gpu/intel-gpu-i915-backports/) for older enterprise / LTS kernel versions, having better support for new HW before upstream kernel does. API it provides to user-space can differ from the eventual upstream version.
+* `xe` is a new KMD that is intended to support future GPUs. While it has [experimental support for latest current GPUs](https://docs.kernel.org/gpu/rfc/xe.html) (starting from Tigerlake), it will not support them officially.
+
+For optimal performance, the KMD should be paired with the same UMD variant. When creating a workload container, depending on the target hardware, the UMD packages should be selected approriately.
+
+| KMD | UMD packages | Support notes |
+|:---- |:-------- |:------- |
+| `i915 upstream` | Distro Repository | For Integrated GPUs. Newer Linux kernels will introduce support for Arc, Flex or Max series. |
+| `i915 backport` | [Intel Repository](https://dgpu-docs.intel.com/driver/installation.html#install-steps) | Best for Arc, Flex and Max series. Untested for Integrated GPUs. |
+| `xe` | Source code only | Experimental support for Arc, Flex and Max series. |
+
+> *NOTE*: Xe UMD is in active development and should be considered as experimental.
+
+Creating a workload that would support all the different KMDs is not currently possible. Below is a table that clarifies how each domain supports different KMDs.
+
+| Domain | i915 upstream | i915 backport | xe | Notes |
+|:---- |:-------- |:------- |:------- |:------- |
+| Compute | Default | [NEO_ENABLE_i915_PRELIM_DETECTION](https://github.com/intel/compute-runtime/blob/3341de7a0d5fddd2ea5f505b5d2ef5c13faa0681/CMakeLists.txt#L496-L502) | [NEO_ENABLE_XE_DRM_DETECTION](https://github.com/intel/compute-runtime/blob/3341de7a0d5fddd2ea5f505b5d2ef5c13faa0681/CMakeLists.txt#L504-L510) | All three KMDs can be supported at the same time. |
+| Media | Default | [ENABLE_PRODUCTION_KMD](https://github.com/intel/media-driver/blob/a66b076e83876fbfa9c9ab633ad9c5517f8d74fd/CMakeLists.txt#L58) | [ENABLE_NEW_KMD](https://github.com/intel/media-driver/blob/a66b076e83876fbfa9c9ab633ad9c5517f8d74fd/media_driver/cmake/linux/media_feature_flags_linux.cmake#L187-L190) | Only one KMD can be enabled at a time. |
```

Comment:
Config option name changed and one can now enable both OOT & Xe KMD support at the same time. IMHO it's also better to link README as it tells which Xe KMD to use:
```suggestion
| Media | Default | [ENABLE_PRODUCTION_KMD](https://github.com/intel/media-driver/blob/a66b076e83876fbfa9c9ab633ad9c5517f8d74fd/CMakeLists.txt#L58) | [ENABLE_XE_KMD](https://github.com/intel/media-driver/blob/intel-media-24.1.4/media_softlet/linux/common/os/xe/include/README.md) | Only support for one i915 KMD variant can be enabled at a time. |
```

Reply:
When you say "config option name changed" what do you mean?
```
if(ENABLE_XE_KMD)
    message("New XE Kmd support has been enabled")
    add_definitions(-DENABLE_XE_KMD)
endif()
```
Link's version seems same to the one in master?

> https://github.com/intel/media-driver/blob/intel-media-24.1.4/media_softlet/linux/common/os/xe/include/README.md

This table is about Xe UMD support. Your link is about the Xe KMD and how the media project's headers are synced to the KMD? I don't see the connection to GPU plugin.

Reply:
> Link's version seems same to the one in master?

The previously linked CMake file gives the correct `ENABLE_XE_KMD` config option name, but the _table itself_ listed the old `ENABLE_NEW_KMD` name.

> This table is about Xe UMD support. Your link is about the Xe KMD and how the media project's headers are synced to the KMD? I don't see the connection to GPU plugin.

Yes.  Because table lists the relevant IMD build config, I would assume it's also relevant which actual Xe KMD uAPI (i.e. Xe KMD repo commit) it expects.   As Xe KMD is still experimental, its uAPI changes from time to time, both in KMD repo, and copies of those uAPI files in user-space repos (media driver does not work e.g. with Xe KMD uAPI from December, or one from Xe KMD repo tip).

Another option would be to just drop the commit/tag related links completely as their info can go stale, and e.g. mention from which version onward, drivers have Xe support:
* mesa (3D/compute/media driver):  23.1.0
* intel-media (media driver): 24.1.4
* compute-runtime (compute/level-zero driver): not tagged / released yet

Reply:
> The previously linked CMake file gives the correct ENABLE_XE_KMD config option name, but the table itself listed the old ENABLE_NEW_KMD name.

Got it, thanks!

> Yes. Because table lists the relevant IMD build config, I would assume it's also relevant which actual Xe KMD uAPI (i.e. Xe KMD repo commit) it expects. As Xe KMD is still experimental, its uAPI changes from time to time, both in KMD repo, and copies of those uAPI files in user-space repos (media driver does not work e.g. with Xe KMD uAPI from December, or one from Xe KMD repo tip).
> 
> Another option would be to just drop the commit/tag related links completely as their info can go stale, and e.g. mention from which version onward, drivers have Xe support

I think that whatever information we have here, will become stale for Xe in a matter of weeks. Thus I'd leave any commits or version details out of here. Once the stack stabilizes we can note that such and such versions are good point to start but before that we'd be just playing catch up all the time.

Also, I'd put that sort of details into its own page.

Reply:
While UMD's Xe KMD support may regress (e.g. with Xe KMD updates), info about first UMD version that added Xe support (option), does not get stale.

I'm fine with just the config option name having been updated, though => resolved.